projects/plugins/crm/src/automation/commons/conditions/quotes/class-quote-status-changed.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * Jetpack CRM Automation Quote_Status_Changed condition.
+ *
+ * @package automattic/jetpack-crm
+ */
+
+namespace Automattic\Jetpack\CRM\Automation\Conditions;
+
+use Automattic\Jetpack\CRM\Automation\Automation_Exception;
+use Automattic\Jetpack\CRM\Automation\Automation_Logger;
+use Automattic\Jetpack\CRM\Automation\Base_Condition;
+
+/**
+ * Quote_Status_Changed condition class.
+ *
+ * @since $$next-version$$
+ */
+class Quote_Status_Changed extends Base_Condition {
+
+	/**
+	 * The Automation logger.
+	 *
+	 * @since $$next-version$$
+	 * @var Automation_Logger $logger The Automation logger.
+	 */
+	private $logger;
+
+	/**
+	 * All valid operators for this condition.
+	 *
+	 * @since $$next-version$$
+	 * @var string[] $valid_operators Valid operators.
+	 */
+	private $valid_operators = array(
+		'is',
+		'is_not',
+	);
+
+	/**
+	 * All valid attributes for this condition.
+	 *
+	 * @since $$next-version$$
+	 * @var string[] $valid_operators Valid attributes.
+	 */
+	private $valid_attributes = array(
+		'operator',
+		'value',
+	);
+
+	/**
+	 * Quote_Status_Changed constructor.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $step_data The step data for the condition.
+	 */
+	public function __construct( array $step_data ) {
+		parent::__construct( $step_data );
+
+		$this->logger = Automation_Logger::instance();
+	}
+
+	/**
+	 * Executes the condition. If the condition is met, the value stored in the
+	 * attribute $condition_met is set to true; otherwise, it is set to false.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $data The data this condition has to evaluate.
+	 * @return void
+	 * @throws Automation_Exception If an invalid operator is encountered.
+	 */
+	public function execute( array $data ) {
+		if ( ! $this->is_valid_quote_status_changed_data( $data ) ) {
+			$this->logger->log( 'Invalid quote status changed data', $data );
+			$this->condition_met = false;
+			return;
+		}
+
+		$field    = 'status';
+		$operator = $this->get_attributes()['operator'];
+		$value    = $this->get_attributes()['value'];
+
+		$this->logger->log( 'Condition: ' . $field . ' ' . $operator . ' ' . $value . ' => ' . $data['data'][ $field ] );
+
+		switch ( $operator ) {
+			case 'is':
+				$this->condition_met = ( $data['data'][ $field ] === $value );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			case 'is_not':
+				$this->condition_met = ( $data['data'][ $field ] !== $value );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+		}
+		$this->condition_met = false;
+		$this->logger->log( 'Invalid operator: ' . $operator );
+
+		throw new Automation_Exception( 'Invalid operator: ' . $operator );
```

Comment:
I have the same suggestion here as the previous PR I reviewed: check this at the beginning of the function using `$valid_operators` and then have a default case in the switch that throws an "unimplemented operator exception".